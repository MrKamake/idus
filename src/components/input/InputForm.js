import React from 'react';
import './InputForm.scss';

const InputForm = ({ type, comment, onChange }) => {
  return (
    <div className="ui_input_form">
      <div className="ui_input_form__textarea">
        <textarea
          className={`ui__textarea ${type}`}
          name="content_text"
          maxLength={500}
          placeholder="주문 요청사항을 입력해주세요"
          value={comment}
          onChange={onChange}
          readOnly={type === 'disabled' || type === 'readonly'}
        ></textarea>
        <em className="ui_textarea__chars">500</em>
      </div>
      <button
        type="submit"
        className={`ui_textarea__btn ${
          type === 'readonly' || (comment && 'active-btn')
        }`}
      >
        저장
      </button>
    </div>
  );
};

export default InputForm;
