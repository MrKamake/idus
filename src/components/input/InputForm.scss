@import '../../styles/utils.scss';

.ui_input_form {
  position: relative;
  margin-top: 20px;
  width: 642px;
  height: 80px;
  overflow: hidden;
  .ui_input_form__textarea {
    position: absolute;
    left: 0;
    right: 0;
    height: 80px;
    .ui__textarea {
      padding: 4px 12px;
      width: 100%;
      height: 100%;
      box-sizing: border-box;
      border: 1px solid $gray8;
      border-radius: 2px;
      line-height: 1.5;
      box-shadow: 0 1px 3px 0 $gray2;
      font-size: 12px;
      resize: none;
      text-indent: 0;
      &::placeholder {
        font-weight: bold;
        color: $idusRedYello;
      }
      &:focus {
        outline: none;
        border: 1px solid $idusGreenBlue;
        border-radius: 2px;
      }
    }
    .disabled {
      pointer-events: none;
      background-color: $transparentGray;
      &::placeholder {
        color: $transparentRed;
      }
    }
    .readonly {
      pointer-events: none;
      background-color: $transparentGray;
      color: $gray9;
    }
    .ui_textarea__chars {
      position: absolute;
      bottom: 10px;
      right: 20px;
      font-size: 12px;
      color: $gray3;
      font-style: normal;
    }
  }
  .active-textarea {
    width: 88%;
  }
  .ui_textarea__btn {
    position: absolute;
    right: 0;
    margin-right: -72px;
    padding: 0 16px;
    width: 72px;
    height: 80px;
    line-height: 78px;
    color: $idusGreenBlue;
    font-size: 12px;
    font-weight: 400;
    background-color: transparent;
    box-shadow: 0 1px 3px 0 $gray2;
    box-sizing: border-box;
    border-radius: 2px;
    cursor: pointer;
    &:hover {
      background-color: $gray10;
    }
  }
  .active-btn {
    margin-right: 0;
  }
}
