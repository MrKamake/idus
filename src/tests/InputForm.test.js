import React from 'react';
import { shallow } from 'enzyme';
import InputForm from '../components/input/InputForm';

describe('InputForm', () => {
  describe('<InputForm />', () => {
    let wrapper, props;

    beforeEach(() => {
      props = {
        type: 'input',
        defaultComment:
          '항상 건강하시고 행복하세요. 그리고 주문시 요청사항 꼭 적어주세요:)',
        text: '',
        onChange: jest.fn(),
        totalByte: 0
      };
      wrapper = shallow(<InputForm {...props} />);
    });

    it('matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should contain elements', () => {
      expect(wrapper.find('textarea').length).toBe(1);
      expect(wrapper.find('button').length).toBe(1);
    });

    it('should not contain node', () => {
      props = {
        type: 'disabled',
        defaultComment: '',
        text: '',
        onChange: jest.fn(),
        totalByte: 0
      };
      wrapper = shallow(<InputForm {...props} />);
      expect(wrapper.find('.active-textarea').length).toBe(0);
      expect(wrapper.find('.active-btn').length).toBe(0);
    });
  });
});
