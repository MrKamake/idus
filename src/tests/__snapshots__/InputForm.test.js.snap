// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm <InputForm /> matches snapshot 1`] = `
<div
  className="ui_input_form"
>
  <div
    className="ui_input_form__textarea "
  >
    <textarea
      className="ui__textarea input"
      maxLength="500"
      name="content_text"
      onChange={[MockFunction]}
      placeholder="주문 요청사항을 입력해주세요"
      readOnly={false}
      value="항상 건강하시고 행복하세요. 그리고 주문시 요청사항 꼭 적어주세요:)"
    />
    <em
      className="ui_textarea__chars"
    >
      434
    </em>
  </div>
  <button
    className="ui_textarea__btn "
    type="submit"
  >
    Save
  </button>
</div>
`;
